harness/determined/keras/_callback.py

```diff
@@ -0,0 +1,446 @@
+    _supports_tf_logs = False
+
+    def __init__(
+        self,
+        core_context: core.Context,
+        checkpoint: Optional[str],
+        continue_id: Union[int, str],
+        *,
+        train_metrics_report_period: Union[int, str] = "epoch",
+        checkpoint_epochs: int = 1,
+    ) -> None:
+        # We subclass ProgbarLogger to disable standard verbose=1 behavior, but really we don't
+        # want any of its actual behavior.  So __init__ the supersuper class directly.
+        callbacks.Callback.__init__(self)
+        self._core = core_context
+        self._checkpoint = checkpoint
+        self._continue_id = continue_id
+        self._report_period = train_metrics_report_period
+        self._checkpoint_epochs = checkpoint_epochs
+
+        self._is_chief = core_context.distributed.rank == 0
+
+        # This is an undocumented workaround in case off-cluster user-code saving goes awry.
+        self._save_user_code = True
+
+        self._steps_completed = 0
+        self._last_ckpt_epoch = -1
+        self._last_train_epoch = -1
+
+        # progress
+        self._training_length: Optional[int] = None
+        self._validation_length: Optional[int] = None
+        self._training_batches = 0
+        self._validation_batches = 0
+        self._percent_reported = -1
+
+        # We download the checkpoint, then have to keep it for a while until we can delete it.
+        self._ckpt_context: Optional[contextlib.ExitStack] = None
+
+    # Mask the inherited ProgbarLogger behavior that we don't actually want.
+    def set_params(self, params: Dict[str, Any]) -> None:
+        callbacks.Callback.set_params(self, params)
+
+    # Mask the inherited ProgbarLogger behavior that we don't actually want.
+    def on_predict_begin(self, logs: Optional[Dict[str, Any]]) -> None:
+        pass
+
+    # Mask the inherited ProgbarLogger behavior that we don't actually want.
+    def on_predict_batch_end(self, batch: int, logs: Optional[Dict[str, Any]]) -> None:
+        pass
+
+    # Mask the inherited ProgbarLogger behavior that we don't actually want.
+    def on_predict_end(self, logs: Optional[Dict[str, Any]]) -> None:
+        pass
+
+    def _implements_train_batch_hooks(self) -> bool:
+        return True
+
+    def _implements_test_batch_hooks(self) -> bool:
+        # Tell keras that we don't need on_test_batch_end unless we are verbose.
+        return bool(self.params["verbose"] != 0)
+
+    def _implements_predict_batch_hooks(self) -> bool:
+        # Tell keras that we don't actually want any on_predict_batch_end calls.
+        return False
+
+    def _print_progress(
+        self, logs: Optional[Dict[str, Any]], training: bool, batches: int, total: Optional[int]
+    ) -> None:
+        # Only report progress if we have the target total.
+        if total is None:
+            return
+
+        # Don't report more often than 10% increments.
+        percent_10 = int((batches / total) * 10) * 10
```

Comment:
nit: could this logic be in the callers, and could it just be based off the batch? like 
```python3
if batch % 10 == 0:
    self._print_progress(...)
```

Reply:
What you wrote prints every 10 batches, and what I have prints every 10 percent of progress, they are not equivalent.

In particular, I recall this was tricky to write (here I copied it from our _DeterminedProgress callback for TFKerasTrial), but it was tricky because handling very many batches per epoch and also very few batches per epoch and making both outputs look good was tricky, which is why I track the percent_10 and percent separately.

I don't see how pulling anything into the caller would be better.

Reply:
> What you wrote prints every 10 batches, and what I have prints every 10 percent of progress, they are not equivalent.

i'm not really sure what i was thinking. like i knew this, but i swear i still thought my comment made sense somehow lol. maybe i thought why bother with the percents, to which your explanation answers.

anyway, i think it's fine what you have. i only wanted to move it out of the method because it seemed like "print progress" should just print progress, instead of doing some checks as to whether it should print progress.